src/main/java/com/org/jenkins/custom/jenkins/distribution/service/services/UpdateCenterService.java

```diff
@@ -25,31 +25,39 @@
     private String responseString;
     private String updateCenterFilePath = "";
 
+
     public JSONObject downloadUpdateCenterJSON() throws Exception {
         /*
         * Check if updateFlag has been set if not then it is the first time the application
         * is being run so we need to download the update-center
         */
         LOGGER.info("Update flag is " + updateFlag);
         LOGGER.info("Update Center Path " + updateCenterFilePath);
-
         if(updateFlag == 0) {
+            downloadJSON(UPDATE_CENTER_JSON_URL);
+            updateFlag = 1;
+        } else {
+            responseString = readFileAsString(updateCenterFilePath);
+        }
+        LOGGER.info("Returning Response" + responseString);
+        return util.convertPayloadToJSON(responseString);
+    }
+
+   public JSONObject downloadJSON (String updateCenterURL) throws Exception {
             File updateCenterFile = File.createTempFile("update-center", ".json");
             updateCenterFilePath = updateCenterFile.getPath();
             LOGGER.info("Creating a new file" + updateCenterFile.getPath());
-            HttpGet get = new HttpGet(UPDATE_CENTER_JSON_URL);
-            LOGGER.info("Executing Request at " + UPDATE_CENTER_JSON_URL);
+            HttpGet get = new HttpGet(updateCenterURL);
+            LOGGER.info("Executing Request at " + updateCenterURL);
             CloseableHttpClient httpClient = HttpClients.createDefault();
             CloseableHttpResponse response = httpClient.execute(get);
             responseString = EntityUtils.toString(response.getEntity());
             byte[] buf = responseString.getBytes(StandardCharsets.UTF_8);
             Files.write(updateCenterFile.toPath(), buf);
-            updateFlag = 1;
-        } else {
-            responseString = readFileAsString(updateCenterFilePath);
-        }
-        LOGGER.info("Returning Response");
-        return util.convertPayloadToJSON(responseString);
+            LOGGER.info("Returning Response");
+            // Mark the file for deletion once the JVM shuts down
+            updateCenterFile.deleteOnExit();
```

Comment:
From [File.deleteOnExit()](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#deleteOnExit--):

> Deletion will be attempted only for normal termination of the virtual machine, as defined by the Java Language Specification.

I do not know if this is a problem or not.

Reply:
Even I am not too sure, but I thought if we do not delete it would just pile up